wrappers/javascript/anoncreds-shared/src/api/Credential.ts

```diff
@@ -1,66 +1,138 @@
+  credentialDefinition: CredentialDefinition | JsonObject
+  revocationRegistryDefinition?: RevocationRegistryDefinition | JsonObject
 }
 
 export class Credential extends AnoncredsObject {
   public static create(options: CreateCredentialOptions) {
-    const credential = anoncreds.createCredential({
-      credentialDefinition: options.credentialDefinition.handle,
-      credentialDefinitionPrivate: options.credentialDefinitionPrivate.handle,
-      credentialOffer: options.credentialOffer.handle,
-      credentialRequest: options.credentialRequest.handle,
-      attributeRawValues: options.attributeRawValues,
-      attributeEncodedValues: options.attributeEncodedValues,
-      revocationRegistryId: options.revocationRegistryId,
-      revocationConfiguration: options.revocationConfiguration?.native,
-      revocationStatusList: options.revocationStatusList?.handle,
-    })
+    let credential
+    // Objects created within this method must be freed up
+    const objectHandles: ObjectHandle[] = []
+    try {
+      const credentialDefinition =
+        options.credentialDefinition instanceof CredentialDefinition
+          ? options.credentialDefinition.handle
+          : pushToArray(CredentialDefinition.fromJson(options.credentialDefinition).handle, objectHandles)
+
+      const credentialDefinitionPrivate =
+        options.credentialDefinitionPrivate instanceof CredentialDefinitionPrivate
+          ? options.credentialDefinitionPrivate.handle
+          : pushToArray(CredentialDefinitionPrivate.fromJson(options.credentialDefinitionPrivate).handle, objectHandles)
+
+      const credentialOffer =
+        options.credentialOffer instanceof CredentialOffer
+          ? options.credentialOffer.handle
+          : pushToArray(CredentialOffer.fromJson(options.credentialOffer).handle, objectHandles)
+
+      const credentialRequest =
+        options.credentialRequest instanceof CredentialRequest
+          ? options.credentialRequest.handle
+          : pushToArray(CredentialRequest.fromJson(options.credentialRequest).handle, objectHandles)
+
+      const revocationStatusList =
+        options.revocationStatusList instanceof RevocationStatusList
+          ? options.revocationStatusList.handle
+          : options.revocationStatusList !== undefined
+          ? pushToArray(RevocationStatusList.fromJson(options.revocationStatusList).handle, objectHandles)
+          : undefined
 
+      credential = anoncreds.createCredential({
+        credentialDefinition,
+        credentialDefinitionPrivate,
+        credentialOffer,
+        credentialRequest,
+        attributeRawValues: options.attributeRawValues,
+        attributeEncodedValues: options.attributeEncodedValues,
+        revocationRegistryId: options.revocationRegistryId,
+        revocationConfiguration: options.revocationConfiguration?.native,
+        revocationStatusList,
+      })
+    } finally {
+      objectHandles.forEach((handle) => handle.clear())
```

Comment:
oh the clearing, ugh haha. 

I am wondering if in the future we can add a feature which removes the need for handles. This means that there might be some performance hits as we only work with raw values, but it might be good for the JS wrapper. 

Something for the future though, this is fine for now ;).